### PR TITLE
Fix for Custom Regions with RedisCacheProvider

### LINKED ISSUE #66: StackExchangeRedis: Custom Regions with RedisCacheProvider
Hi @maca88 ,
Because of "RedisCacheProvider.ConfiguredCacheRegions" is private readonly  I cannot specify region based expiration with hibernate config file, give us a way to implement custom configuration.
If we want to write our RedisCacheProvider we are nearly copying this project.

Below config option is not enough with .net core add support for region configuration like "ConfigureRegions(RegionConfig[] regions)" 
> NHibernate.Caches.StackExchangeRedis
```c#
        static RedisCacheProvider()
        {
            Log = NHibernateLogger.For(typeof(RedisCacheProvider));
            ConfiguredCacheRegions = new Dictionary<string, RegionConfig>();

            if (!(ConfigurationManager.GetSection("redis") is CacheConfig config))
                return;

            ConfiguredCache = config;
	    foreach (var cache in config.Regions)
	    {
			ConfiguredCacheRegions.Add(cache.Region, cache);
	    }
        }
```

Also if we specify "cache.region_prefix" in hibernate configuration file it is added twice to key.
One is added in Nhibernate code the other one is NHibernate.Caches.StackExchangeRedis;

> NHibernate
```c#
                internal string GetFullCacheRegionName(string name)
		{
			var prefix = CacheRegionPrefix;
			if (!string.IsNullOrEmpty(prefix))
				return prefix + '.' + name;
			return name;
		}
```

> NHibernate.Caches.StackExchangeRedis
```c#
        private void BuildDefaultConfiguration(IDictionary<string, string> properties)
        {
            var config = CacheConfiguration;

            config.CacheKeyPrefix = GetString(RedisEnvironment.KeyPrefix, properties, config.CacheKeyPrefix);
            Log.Debug("Cache key prefix: {0}", config.CacheKeyPrefix);

            config.EnvironmentName = GetString(RedisEnvironment.EnvironmentName, properties, config.EnvironmentName);
            Log.Debug("Cache environment name: {0}", config.EnvironmentName);

            config.RegionPrefix = GetString(Cfg.Environment.CacheRegionPrefix, properties, config.RegionPrefix);
            // if region name contains region prefix so do not add to RegionPrefix, because it's triggered from Nhibernate code base?
        .
        .
        .
        }
```

### PR DESCRIPTION
**Pull request by maca88**
Fixes #66

### CODE REVIEW

File: StackExchangeRedis/NHibernate.Caches.StackExchangeRedis/RedisCacheProvider.cs [hunk 1]
```diff
@@ -28,6 +28,16 @@ public static RedisCacheConfiguration DefaultCacheConfiguration
 			set => _defaultCacheConfiguration = value ?? new RedisCacheConfiguration();
 		}
 
+		/// <summary>
+		/// Set the region configuration.
+		/// </summary>
+		/// <param name="regionName">The region name.</param>
+		/// <param name="configuration">The region configuration.</param>
+		public static void SetRegionConfiguration(string regionName, RegionConfig configuration)
```
Comment (gokhanabatay) [inline StackExchangeRedis/NHibernate.Caches.StackExchangeRedis/RedisCacheProvider.cs]: To allow override of virtual methods please make _connectionMultiplexer protected.

Comment (gokhanabatay) [reply StackExchangeRedis/NHibernate.Caches.StackExchangeRedis/RedisCacheProvider.cs]: Could you "SetRegionConfiguration" method to other cache providers?

Comment (maca88) [reply StackExchangeRedis/NHibernate.Caches.StackExchangeRedis/RedisCacheProvider.cs]: > please make _connectionMultiplexer protected

It was intentionally made private as the derivered type may use multiple connections (check [DistributedRedisCacheProvider](https://github.com/nhibernate/NHibernate-Caches/blob/790b3841daf12859a2635d82dd96840168fff85b/StackExchangeRedis/NHibernate.Caches.StackExchangeRedis.Tests/Providers/DistributedRedisCacheProvider.cs)), but as such case is rare, I've still made it protected.

> Could you "SetRegionConfiguration" method to other cache providers?

Will do it in a separate PR.

**Comment by maca88** [comment]
Moved `SetRegionConfiguration` method to #68.

**Pull request merged**